includes/preflight.inc

```diff
@@ -441,7 +441,7 @@ function _drush_preflight_columns() {
   if (!($columns = getenv('COLUMNS') ?: 0)) {
     // Trying to export the columns using stty.
     exec('stty size 2>&1', $columns_output, $columns_status);
-    if (!$columns_status) $columns = preg_replace('/\d+\s(\d+)/', '$1', $columns_output[0], -1, $columns_count);
+    if (!$columns_status && is_numeric($columns_output[0])) $columns = preg_replace('/\d+\s(\d+)/', '$1', $columns_output[0], -1, $columns_count);
 
     // If stty fails and Drush us running on Windows are we trying with mode con.
     if (($columns_status || !$columns_count) && drush_is_windows()) {
```

Comment:
I think it would be better to fix the problem here, as I can imagine instances where is_numeric is true, but the preg_replace still does not work correctly. Just change this so that default behavior happens when $columns comes out as zero (or less).

Reply:
What about lines 459-461? We could maybe change it into:
```
     if (empty($columns) || !is_numeric($columns)) {
       $columns = 80;
     }
```
That would cover all wrong values of $columns I think.